internal/http/services/owncloud/ocs/handlers/apps/sharing/shares/remote.go

```diff
@@ -82,77 +70,61 @@ func (h *Handler) createFederatedCloudShare(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
-	var permissions conversions.Permissions
-	var role string
+	var role *conversions.Role
 
 	pval := r.FormValue("permissions")
 	if pval == "" {
 		// by default only allow read permissions / assign viewer role
-		permissions = conversions.PermissionRead
-		role = conversions.RoleViewer
+		role = conversions.NewViewerRole()
 	} else {
 		pint, err := strconv.Atoi(pval)
 		if err != nil {
 			response.WriteOCSError(w, r, response.MetaBadRequest.StatusCode, "permissions must be an integer", nil)
 			return
 		}
-		permissions, err = conversions.NewPermissions(pint)
+		permissions, err := conversions.NewPermissions(pint)
 		if err != nil {
 			response.WriteOCSError(w, r, response.MetaBadRequest.StatusCode, err.Error(), nil)
 			return
 		}
-		role = conversions.Permissions2Role(permissions)
+		role = conversions.RoleFromOCSPermissions(permissions)
 	}
 
-	var resourcePermissions *provider.ResourcePermissions
-	resourcePermissions, err = h.map2CS3Permissions(role, permissions)
-	if err != nil {
-		log.Warn().Err(err).Msg("unknown role, mapping legacy permissions")
-		resourcePermissions = asCS3Permissions(permissions, nil)
-	}
-
-	statReq := &provider.StatRequest{
-		Ref: &provider.Reference{
-			Spec: &provider.Reference_Path{
-				Path: path.Join(prefix, r.FormValue("path")),
-			},
-		},
-	}
-	statRes, err := c.Stat(ctx, statReq)
-	if err != nil {
-		response.WriteOCSError(w, r, response.MetaServerError.StatusCode, "error sending a grpc stat request", err)
-		return
-	}
-	if statRes.Status.Code != rpc.Code_CODE_OK {
-		if statRes.Status.Code == rpc.Code_CODE_NOT_FOUND {
-			response.WriteOCSError(w, r, response.MetaNotFound.StatusCode, "not found", nil)
-			return
-		}
-		response.WriteOCSError(w, r, response.MetaServerError.StatusCode, "grpc stat request failed", err)
-		return
+	if statInfo != nil && statInfo.Type == provider.ResourceType_RESOURCE_TYPE_FILE {
+		// Single file shares should never have delete or create permissions
+		permissions := role.OCSPermissions()
+		permissions &^= conversions.PermissionCreate
+		permissions &^= conversions.PermissionDelete
```

Comment:
yeah, the conversions package needs to take into account the file type. so it can calculate the correct set of ocs permissions and cs3 permission flags for a role. As the comment says files never have an OCS Create and Delete permission.